src/ZkFold/Base/Algebra/Basic/Class.hs

```diff
@@ -249,24 +514,30 @@ instance Ring a => Ring [a]
 
 --------------------------------------------------------------------------------
 
-instance AdditiveSemigroup a => AdditiveSemigroup (p -> a) where
-    p1 + p2 = \x -> p1 x + p2 x
-
-instance AdditiveMonoid a => AdditiveMonoid (p -> a) where
-    zero = const zero
-
-instance AdditiveGroup a => AdditiveGroup (p -> a) where
-    negate = fmap negate
-
 instance MultiplicativeSemigroup a => MultiplicativeSemigroup (p -> a) where
     p1 * p2 = \x -> p1 x * p2 x
 
+instance Exponent b a => Exponent b (p -> a) where
+    f ^ p = \x -> f x ^ p
+
 instance MultiplicativeMonoid a => MultiplicativeMonoid (p -> a) where
     one = const one
 
 instance MultiplicativeGroup a => MultiplicativeGroup (p -> a) where
     invert = fmap invert
 
+instance AdditiveSemigroup a => AdditiveSemigroup (p -> a) where
+    p1 + p2 = \x -> p1 x + p2 x
+
+instance Scale b a => Scale b (p -> a) where
```

Comment:
I suppose we would get this for free if Scale is defined on Functors

Reply:
Yeah, definitely